ebu_tt_live/node/denester.py

```diff
@@ -22,280 +32,277 @@ def denest(document):
-        return div_attributes
     
-    @staticmethod
-    def merge_attr(parent_attr, div_attributes):
-        merged_attributes ={ "styles": [], "begin": None, "end": None, "lang": None, "region": None }
-
-        if not isinstance(parent_attr["begin"], timedelta) and parent_attr["begin"] is not None:
-            parent_attr["begin"] = parent_attr["begin"].timedelta
-        if not isinstance(parent_attr["end"], timedelta) and parent_attr["end"] is not None:
-            parent_attr["end"] = parent_attr["end"].timedelta
-        if not isinstance(div_attributes["begin"], timedelta) and div_attributes["begin"] is not None:
-            div_attributes["begin"] = div_attributes["begin"].timedelta
-        if not isinstance(div_attributes["end"], timedelta) and div_attributes["end"] is not None:
-            div_attributes["end"] = div_attributes["end"].timedelta
-
-        if div_attributes["styles"] is not None:
-             merged_attributes["styles"] = div_attributes["styles"] + parent_attr["styles"]
-        else:
-            merged_attributes["styles"] = parent_attr["styles"]
-        if div_attributes["lang"] is not None:
-            merged_attributes["lang"] = div_attributes["lang"]
-        else:
-            merged_attributes["lang"] = parent_attr["lang"]
-        if parent_attr["region"] is not None:
-            merged_attributes["region"] = parent_attr["region"]
-        elif div_attributes["region"] is not None:
-            merged_attributes["region"] = div_attributes["region"]
-        if parent_attr["begin"] is not None and div_attributes["begin"] is not None:
-            merged_attributes["begin"] = parent_attr["begin"] + div_attributes["begin"]
-        elif parent_attr["begin"] is not None and div_attributes["begin"] is None:
-            merged_attributes["begin"] = parent_attr["begin"]
-        elif parent_attr["begin"] is None and div_attributes["begin"] is None:
-            merged_attributes["begin"] = None
-        else:
-            merged_attributes["begin"] = div_attributes["begin"] if parent_attr["begin"] is None else parent_attr["begin"]
-        if parent_attr["end"] is not None and div_attributes["end"] is not None:
-            merged_attributes["end"] = parent_attr["end"] - div_attributes["end"]
-        elif parent_attr["end"] is not None and div_attributes["end"] is None:
-            merged_attributes["end"] = parent_attr["end"]
-        else:
-            merged_attributes["end"] = Denester.calculate_end_times(parent_attr, div_attributes, parent_attr["begin"])
-        return merged_attributes
-
-    @staticmethod
-    def calculate_end_times(parent_attr, child_attr, time_sync):
-        if child_attr["end"] is not None and parent_attr["end"] is None and time_sync is not None:
-           return time_sync+child_attr["end"]
-        elif parent_attr["end"] is not None and child_attr["end"] is not None:
-             return parent_attr["end"]-child_attr["end"]
+def check_p_regions(divs):
```

Comment:
The indentation has dropped out here - are these intended to be global functions now?